template/src/App.tsx

```diff
@@ -1,22 +1,29 @@
 import * as React from 'react';
-import './App.css';
-
+import * as styles from './App.css';
 import logo from './logo.svg';
 
 class App extends React.Component {
   public render() {
     return (
-      <div className="App">
-        <header className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <h1 className="App-title">Welcome to React</h1>
+      <div className={styles.app}>
+        <header className={styles.header}>
+          <img src={logo} className={styles.logo} alt="logo" />
+          <p>
+            Edit <code>src/App.tsx</code> and save to reload.
+          </p>
+          <a
+            className={styles.link}
+            href="https://reactjs.org"
+            target="_blank"
+            rel="noopener noreferrer"
+          >
+            Learn React
+          </a>
         </header>
-        <p className="App-intro">
-          To get started, edit <code>src/App.tsx</code> and save to reload.
-        </p>
       </div>
     );
   }
 }
 
 export default App;
```

Comment:
Same question for the reason of the difference here.
Additionally, CSS modules are a bit more complicated to use in TS without a proper type declaration. There are special loaders for this, like: https://github.com/olegstepura/typed-css-modules-loader
But apparently, this causes additional overhead to integrate properly.

Reply:
I think it would be in the spirit of this package to provide a css modules loader. Since CRA provides a CSS Modules loader in its config by default, this would be part of maintaining feature parity, but with TS. I tried [this one](https://github.com/Jimdo/typings-for-css-modules-loader), but it didn't seem to generate the correct typings or at least I couldn't figure it out in the time I had. The important thing is that the regexes for modules be separate, like `.module.css` and `.module.scss`, so that users using regular SASS or CSS don't experience the module loader's overhead.

Reply:
A question above is just about the example App code, which differs from CRA.

In the current state of the config, it would be possible to use CSS modules (using the extensions you mentioned), it's just that specific typings would be generated; instead, contents of these files will be typed as a generic `any` for now. I'm not that familiar with the available loaders or plugins to achieve more specific ones at the moment, esp. what they generate and where. However, it's work a closer look and see if we can get this working.

Reply:
Makes sense, applies to #375

Reply:
I highly do *not* recommend adding `typed-css-modules-loader` or any css typing support to this. I've gone down this road recently and only hit hell when dealing with CSS being imported from 3rd party node_modules that have special characters that will make any typing parser spectacularly explode. It's better to leave css loaders as typed as `any` imho.

Reply:
@jadbox Let's discuss on #375 above.